src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -972,8 +983,10 @@ std::vector<std::shared_ptr<const Table>> ExpressionEvaluator::_evaluate_subquer
     _materialize_segment_if_not_yet_materialized(parameter.second);
   }
 
+  // Ensure we are the only ones to execute the input PQP for correlated subqueries and set their parameters.
+  Assert(expression.pqp->state() == OperatorState::Created,
+         "ExpressionEvaluator manages operators of correlated subqueries.");
```

Comment:
@Bouncner do we want to `DebugAssert` this for each operator in the PQP and leave the `Assert` for the root operator?

Reply:
Reading it, I think the message could be improved to state that "ONLY the EE should manage or so". Does not sound like an error message.

Concerning the question: I am not really sure. I lack understanding if a DebugAssert would suffice it. But since the check is basically for free (compared to the other stuff being done), I would probably keep it an Assert.

I guess we need to talk about your question in person. You must explain that to me first. :D